raster-foundry/raster-foundry
# Support lab analysis result async export

## Overview

This PR supports lab analysis result async export.

### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any new SQL strings have tests~

### Demo

Select an area to export analysis result

![video-play-close](https://raw.githubusercontent.com/aaronxsu/MyData/master/gifs/rf-pr-gifs/analysis-aoi-export.gif)

Dropbox

![video-play-close](https://raw.githubusercontent.com/aaronxsu/MyData/master/gifs/rf-pr-gifs/analysis-export-dropbox.gif)

Analysis export download

![video-play-close](https://raw.githubusercontent.com/aaronxsu/MyData/master/gifs/rf-pr-gifs/analysis-export-download.gif)

### Note

Thanks @designmatty for updating the icons.

This PR has been re-scoped from supporting synchronous to asynchronous export.


## Testing Instructions

 * Go to lab, create an analysis, update project and band parameters
 * Open up the lab map view and use the quick export control on the map to select a bounding box to export on your current zoom level.
 * Open/close map control, click on buttons in the control popup, draw rectangles, etc under different circumstances and see if they behave as you expect.
 * Go to analyses list page and check if analyses with exports created are identified and the download modal is as expected

Closes https://github.com/raster-foundry/raster-foundry/issues/2791
